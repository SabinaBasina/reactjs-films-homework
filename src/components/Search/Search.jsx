import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './Search.scss';

class Search extends Component {
  tvInput = React.createRef();

  findTv = () => {
    const { onFindTv } = this.props;
    onFindTv(this.tvInput.current.value);
    this.tvInput.current.value = '';
  }

  render() {
    return (
      <div className="Search">
        <input
          className={styles.InputSearch}
          type="text"
          ref={this.tvInput}
          onKeyPress={(e) => {
            if (e.key === 'Enter') {
              this.findTv();
            }
          }}
          placeholder="Search"
        />
      </div>
    );
  }
}

export default Search;

Search.propTypes = {
  onFindTv: PropTypes.func,
};

Search.defaultProps = {
  onFindTv: () => { },
};
