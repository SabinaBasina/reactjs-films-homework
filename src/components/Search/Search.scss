.InputSearch {
  margin-bottom: 15px;
  margin-top: 30px;
  width: 300px;
  height: 28px;
  padding-left: 15px;
  border-radius: 42px;
  border: 2px solid #222;
  background: rgb(255, 255, 255);
  color: black;
  font-size: 15px;
}
