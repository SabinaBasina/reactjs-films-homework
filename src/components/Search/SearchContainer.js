import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import Search from './Search';
import { getSearchResult } from '../../modules/moduleHome/moduleHomeActions';
import { getValue } from '../../modules/moduleHome/moduleHomeSelectors';


const mapStateToProps = createStructuredSelector({
  value: getValue,
});

const mapDispatchToProps = {
  onFindTv: value => (dispatch) => {
    dispatch({ type: 'FIND_TV', payload: value });
    dispatch(getSearchResult());
  },
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);
