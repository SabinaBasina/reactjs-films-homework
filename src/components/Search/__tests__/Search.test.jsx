import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';

import Search from '../Search';

describe('Search', () => {
  it('should change value', () => {
    const component = ReactTestUtils.renderIntoDocument(<Search />);
    const input = ReactTestUtils.findRenderedDOMComponentWithClass(component, 'InputSearch');
    expect(input.value).toEqual('');
    ReactTestUtils.Simulate.change(input, { target: { value: 'a' } });
    expect(input.value).toEqual('a');
  });

  it('should invoke history.push', () => {
    const historyMock = { push: jest.fn() };
    const component = ReactTestUtils.renderIntoDocument(<Search history={historyMock} />);
    const input = ReactTestUtils.findRenderedDOMComponentWithClass(component, 'InputSearch');
    ReactTestUtils.Simulate.keyPress(input, { key: 'Enter', keyCode: 13, which: 13 });
    expect(historyMock.push.mock.calls.length).toBe(1);
  });

  it('should not invoke history.push', () => {
    const historyMock = { push: jest.fn() };
    const component = ReactTestUtils.renderIntoDocument(<Search history={historyMock} />);
    const input = ReactTestUtils.findRenderedDOMComponentWithClass(component, 'InputSearch');
    ReactTestUtils.Simulate.keyPress(input, { key: 'Space', keyCode: 32, which: 32 });
    expect(historyMock.push).toHaveBeenCalledTimes(0);
  });
});
