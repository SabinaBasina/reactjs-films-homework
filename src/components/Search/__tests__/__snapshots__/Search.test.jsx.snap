// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search test 1`] = `
<div>
  <input
    className="InputSearch"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
